File: java/CodeRetreatTest.java
// Use JUnit 4 and Hamcrest

import org.junit.Test;

import static org.junit.Assert.*;

public class CodeRetreatTest {
  @Test
  public void testNotPalindrome() {
    assertFalse("isPalindrome: notracecar", CodeRetreat.isPalindrome("notracecar"));
  }

  @Test
  public void testIsPalindrome() {
    assertTrue("isPalindrome: kayak", CodeRetreat.isPalindrome("kayak"));
  }

  @Test
  public void testFindMinEmpty() {
    assertEquals("Find minimum of empty array", -1, CodeRetreat.findMinimum(new int[]{}));
  }

  @Test
  public void testFindMinimumNonEmpty() {
    assertEquals("Find minimum in [9, -5, 1, 4]", -5, CodeRetreat.findMinimum(new int[]{9, -5, 1, 4}));
  }
}

File: python/coderetreat_test.py
import coderetreat
import unittest

class CodeRetreat(unittest.TestCase):
  def test_palindrome(self):
    self.assertFalse(coderetreat.is_palindrome('notracecar'))
    self.assertTrue(coderetreat.is_palindrome('kayak'))

  def test_find_min(self):
    self.assertEqual(-1, coderetreat.find_min([]), "Find minimum of empty array")
    self.assertEqual(-5, coderetreat.find_min([9, -5, 1, 4]), "Find minimum in [9, -5, 1, 4]")

if __name__ == '__main__':
  unittest.main()
